<html>
	<head>
		<title>Predicting Protein-Protein Interaction via SVM Learning</title>
		<link rel="stylesheet" href="http://twitter.github.com/bootstrap/1.4.0/bootstrap.min.css" />
		<link rel="stylesheet" href="/css/protein.css" />
	</head>
	<body>
		<div class="topbar" data-scrollspy="scrollspy">
			<div class="topbar-inner">
				<div class="container">
					<a class="brand" href="#overview">Protein Interaction</a>
					<ul class="nav">
						<li><a href="#intro">Introduction</a></li>
						<li><a href="#dataset">Dataset</a></li>
						<li><a href="#fv">Feature vector</a></li>
						<li><a href="#results">Results</a></li>
					</ul>
				</div>
			</div>
		</div>
		<header class="jumbotron masthead" id="overview">
			<div class="inner">
				<div class="container">
					<h1>Predicting Protein-Protein Interaction via SVM Learning</h1>
					<p class="lead">
						<a href="mailto:CharlestonNoble2012@u.northwestern.edu">Charleston Noble</a> (B.A Bioinformatics & Mathematics, 2012)
						<br/>
						<a href="mailto:azylman@u.northwestern.edu">Alex Zylman</a> (B.S. Computer Engineering, 2012)
					</p>
					<p>
						EECS 349: Machine Learning - Fall 2011<br/>
						Bryan Pardo<br/>
						Northwestern University</br>
					</p>
				</div>
		</header>
		<div class="quickstart">
			<div class="container">
				<div class="row">
					<div class="span5">
						<h6>Source code</h6>
						<br/>
						<p>
							<a target="_blank" href="https://github.com/azylman/Protein-Interactions" class="btn primary">Available on GitHub</a>
						</p>
					</div>
					<div class="span5">
						<h6>Poster</h6>
						<br/>
						<p>
							<a target="_blank" href="https://github.com/azylman/protein-website/blob/master/poster/poster_landscape.pdf?raw=true" class="btn primary">Available as a PDF</a>
						</p>
					</div>
					<div class="span5">
						<h6>Abstract</h6>
						<br/>
						<p>
							<a target="_blank" href="blah blah" class="btn primary">Available as a PDF</a>
						</p>
					</div>
				</div>
			</div>
		</div>
		<div class="container">
			<section id="intro">
				<div class="page-header"><h1>Introduction</h1></div>
				<div class="grid">
					<div class="row">
						<div class="span16"></div>
					</div>
					<br />
					<div class="row">
						<div class="span10">
							Interactions between proteins are essential for the large majority of biological functions, so predicting whether two proteins will interact is a central problem in medical and biological research. We propose a method which accomplishes this task with only the most basic knowledge of the two proteins: their primary structures.
							<br/><br/>
							Proteins are linear chains of <strong>amino acids</strong>, which fold to form complex three dimensional structures. A protein's <strong>primary structure</strong> is simply its amino acid sequence, while secondary, tertiary and quaternary structures refer to its local and global three-dimensional shape.
						</div>
						<div class="span6"><img src="/images/protein_structure_fixed-small.png" /></div>
					</div>
					<br />
					<div class="row">
						<div class="span6"><img src="/images/docking-small.png" /></div>
						<div class="span10">If two proteins come into contact with each other, and their structures complement each other in some region, then they can <strong>bind</strong>, or <strong>dock</strong> with each other. If two proteins are capable of binding with each other, then they are said to <strong>interact</strong>. Binding interactions ubiquitous in biological processes, and much of contemporary pharmaceutical research is concerned with the creation of novel interacting proteins. For example, Alzheimer's Disease is thought to be caused by the accumulation of misfolded proteins in the brain, and one proposed cure is the creation of a protein which can interact with these and properly re-orient them.</div>
					</div>
					<br />
					<div class="row">
						<div class="span16">While the interaction (or non-interaction) status of protein pairs is determined by their three dimensional shapes, the problem of accurately predicting the shape of novel proteins remains an open (and enormously complex) problem in bioinformatics. Thus it has been suggested that, since a protein's primary structure determines its tertiary structure, or three dimensional shape, we can skip the intermediate and intractable step of determining the proteins' tertiary structure when attempting to predict interactions.</div>
					</div>
				</div>
			</section>
		
			<section id="dataset">
				<div class="page-header"><h1>Dataset</h1></div>
				<div class="grid">
					<div class="row">
						<div class="span10">
							Primary structures of interacting protein pairs were obtained from UCLA's Database of Interacting Proteins which contains over 50,000 entries representing pairs of proteins known to mutually bind ("positive interactions"). Pairs of proteins known to be unable to bind ("negative interactions") are not included in the database, so we needed to create them by permuting sequences of positively interacting pairs. If one member of a pair of positively interacting proteins is shuffled, the result is a synthetic native-like protein that, with a high degree of certainty, can be said to interact negatively. This permutation was performed using Shufflet, a bioinformatics application developed by Eivind Coward, University of Bergen. Overall, 134,550 protein-pairs were used, with half of these being interacting pairs and the other half non-interacting.
						</div>
						<div class="span6"><img src="/images/interaction_network-small.png" /></div>
					</div>
				</div>
			</section>
			
			<section id="fv">
				<div class="page-header"><h1>Feature vector</h1></div>
				Proteins were represented by feature vectors by aggregating descriptors of their amino acid constituents. The descriptors we considered were <strong>hydrophobicity</strong>, <strong>hydrophicility</strong>, <strong>volume</strong>s of side chains, <strong>polarity</strong>, <strong>polarizability</strong>, solvent-accessible surface area (<strong>SASA</strong>) and net charge index (<strong>NCI</strong>). These descriptors were chosen because they, in particular, affect the binding properties of the overall protein. They have a known value for each amino acid that we normalized before constructing our feature vector.
				<br/><br/>
				Aggregate descriptor values for proteins were assembled by finding the autocovariance (AC) of their values up to a distance lg away from a particular amino acid. Given a protein sequence, AC variables describe the average interactions between residues, a certain lag apart throughout the whole sequence. These values were calculated according to:
				<br/><br/>
				<img src="/images/acc.png" />
				<br/><br/>
				where j represents one descriptor, i the position in the sequence X, n the length of the sequence X and lag <= lg the value of the lag. The software SVM-Light was employed in the classiffication of feature vectors. A linear function was chosen as the kernel function due to time constraints.
			</section>
			
			<section id="results">
				<div class="page-header"><h1>Results</h1></div>
				<div class="grid">
					<div class="row">
						<div class="span10">
							We achieved a 72.56% accuracy rate using a linear kernel and the full set of features. This compares to the roughly 80% accuracy of current research into this problem. From there, we wanted to find out which acid descriptors had the largest effect in determining the accuracy of our classiffier.
							<br/><br/>
							Initially, we ran our classiffier seven times, each time using all of the descriptors except for one. This way, we get an average accuracy of 72.07% (with a standard deviation of .31%), only slightly less than our accuracy using all of the descriptors. The removal of SASA had the largest effect, dropping our accuracy to 71.67%, while the removal of polarizability had the least effect, dropping our accuracy to 72.46%.
						</div>
						<div class="span6"><img src="/images/accuracy-small.png"/></div>
					</div>
					<br/>
					<div class="row">
						<div class="span6"><img src="/images/accuracy_by_num_features-small.png" /></div>
						<div class="span10">Subsequently, we ran our classiffier seven times, each using only one of our seven descriptors. The average accuracy of 61.84% with a standard deviation of 4.79%. The descriptor that had the least effect on accuracy was NCI, which had an accuracy of 57.04%. The descriptor that had the largest effect was polarity, which had an accuracy of 68.08%. While most of our training was done with a linear kernel, we ran this training data a second time using a radial basis kernel function, which increased our accuracy to 71.4%.</div>
					</div>
					<br/>
					<div class="row">
						<div class="span16">Next, we wanted to see which descriptor, when paired with polarity, gave the highest accuracy so we ran our classiffier six times using each of the remaining six descriptors and polarity. We got an average accuracy of 69.32% with a standard deviation of .38%. The descriptor that had the highest accuracy when paired with polarity was SASA at 69.88%, while the descriptor with the lowest accuracy was 68.92%.</div>
					</div>
					<br/>
					<div class="row">
						<div class="span16">When we look at our descriptors paired with polarity and SASA, the average is 70.67% and the standard deviation is .46%. The maximum accuracy of 71.47% was achieved using volume, polarity, and SASA while the lowest accuracy of 70.36% was a result of using polarity, SASA, and NCI.</div>
					</div>
				</div>
			</section>
		</div>
		
		<footer class="footer">
			<div class="container">
				<p class="pull-right">
					<a href="#">Back to top</a>
				</p>
				<p>
					&copy;2011 Alex Zylman and Chuck Noble<br/>
				</p>
			</div>
		</footer>
		
		<script src="http://code.jquery.com/jquery-1.5.2.min.js"></script>
		<script src="http://twitter.github.com/bootstrap/1.4.0/bootstrap-scrollspy.js"></script>
		
	</body>
</html>